automation/features/step-definitions/staking.steps.js

```diff
@@ -1,45 +1,65 @@
 const { When, Then } = require("@wdio/cucumber-framework");
+const stakingPage = require("../pageobjects/staking.page");
 let currentAmountOfTokensInWallet = "";
+let validatorName = "";
+/* eslint-disable  no-unused-vars */
+let tokensAssociatedInVegaWallet = "";
+let tokensAssociatedInVegaWalletText = "";
+let tokensNotAssociatedInVegaWalletText = "";
+/* eslint-disable  no-unused-vars */
+let tokensNotAssociatedInVegaWallet = "";
 
-Then(/^I can see the validator node list$/, () => {
-  const nodeList = $(".node-list").$$('[data-testid="node-list-item"]');
-  expect(nodeList.length).toBeGreaterThan(0);
-});
+When(
+  /^I associate "([^"]*)?" tokens from "([^"]*)?"$/,
+  (tokenAmount, sourceOfFunds) => {
+    /* eslint-disable  wdio/no-pause */
+    browser.pause(500);
+    tokensAssociatedInVegaWalletText = $$(
+      '[data-testid="associated-token-amount"]'
+    )[2].getText();
+    tokensNotAssociatedInVegaWalletText = $$(
+      '[data-testid="not-associated-token-amount"]'
+    )[2].getText();
+    tokensAssociatedInVegaWallet = tokensAssociatedInVegaWalletText.replace(
+      ",",
+      ""
+    );
+    tokensNotAssociatedInVegaWallet =
+      tokensNotAssociatedInVegaWalletText.replace(",", "");
+    stakingPage.associateTokens(tokenAmount, sourceOfFunds);
+  }
+);
 
-When(/^I associate some tokens from wallet$/, () => {
-  browser.getByTestId("callout").waitForDisplayed({
-    timeout: 30000,
-    reverse: false,
-    timeoutMsg: "callout was not found",
-  });
-  if (browser.getByTestId("associate-more-tokens-btn").isDisplayed()) {
-    browser.getByTestId("associate-more-tokens-btn").click();
-  } else browser.getByTestId("associate-tokens-btn").click();
-  $('[data-testid="associate-radio-wallet"]').click({ force: true });
-  // eslint-disable-next-line wdio/no-pause
-  browser.pause(1000);
-  browser.getByTestId("token-amount-input").setValue(0.00001);
-  // eslint-disable-next-line wdio/no-pause
-  browser.pause(1000);
-  browser.getByTestId("token-input-submit-button").click();
-  browser.waitUntil(() => browser.getWindowHandles().length > 1);
-  browser.switchWindow("MetaMask Notification");
-  $("button=Confirm").click();
-  browser.switchWindow("VEGA");
-  // eslint-disable-next-line wdio/no-pause
-  browser.pause(2000);
-});
-
-When(/^I can see the pending transactions button is shown$/, async () => {
-  browser.getByTestId("pending-transactions-btn").waitForDisplayed({
+When(
+  /^I disassociate "([^"]*)?" tokens from "([^"]*)?"$/,
+  (tokenAmount, sourceOfFunds) => {
+    /* eslint-disable  wdio/no-pause */
+    browser.pause(500);
```

Comment:
Either remove or add a comment why this is needed.

Reply:
this pause has been removed